javaslang/src/test/java/javaslang/collection/IntMap.java

```diff
@@ -45,21 +45,17 @@ public boolean isDistinct() {
     }
 
     @Override
-    public int hashCode() {
-        return original.values().hashCode();
-    }
-
-    @Override
     public boolean equals(Object o) {
+        Object map = o;
         if (o instanceof IntMap) {
-            final IntMap<?> that = (IntMap<?>) o;
-            return original.equals(that.original) || original.values().equals(that.original.values());
-        } else if (o instanceof Iterable) {
-            final Iterable<?> that = (Iterable<?>) o;
-            return original.values().equals(that);
-        } else {
-            return false;
+            map = ((IntMap) o).original;
         }
+        return Collections.equals(this.original, map);
```

Comment:
I'm afraid your solution will be rejected by tests.
Yes, `IntMap` and `IntMultimap` taken from hell. All tests belonging to the `Map` and `Multimap` should be rewritten without using these classes, but this is huge work.